### PR TITLE
TG-382 Add redirection and consent form

### PR DESCRIPTION
**Pull request by JeremieSamson**
Capture du 2019-07-30 00-14-41

### CODE REVIEW

File: templates/User/consent.html.twig [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{% extends 'base.html.twig' %}
+
+{% block body %}
+    <div class="container management-form white-box p-3 mt-5 mb-5">
+
+        <div class="text-left">
+            {# Account #}
```
Comment (victor-paumier) [inline templates/User/consent.html.twig]: no comment

File: src/EventListener/UserListener.php [hunk 2]
```diff
@@ -148,31 +149,8 @@ public function prePersist(LifecycleEventArgs $args)
...
-        Email: {$user->getEmail()}
-        Local: {$user->isLocal()}
-        Exchange city: {$city}
-        University: {$university}
-        Semester: {$user->getSemester()}
-        Languages: {$languages}
-EOF;
-
-        $user->setPrivacyAcceptedWith($acceptedWith);
+        if ($user->isPrivacyAccepted()) {
```
Comment (victor-paumier) [inline src/EventListener/UserListener.php]: exec on each user update?

File: src/Controller/ProfileController.php [hunk 3]
```diff
@@ -102,6 +103,33 @@ public function delete(
         ]);
     }
 
+    /**
+     * @Route("/consent", name="user_consent", methods={"GET", "POST"}, options={"expose" : true})
```
Comment (victor-paumier) [inline src/Controller/ProfileController.php]: expose true not needed

File: src/EventListener/UserListener.php (hunk 2 above)
Comment (JeremieSamson) [reply src/EventListener/UserListener.php]: it's under the `prePersist` event

**Pull request merged**